lua/octo/init.lua

```diff
@@ -408,6 +409,48 @@ function M.load_issue()
   )
 end
 
+-- This function accumulates all the taggable users into a single list that
+-- gets set as a buffer variable `taggable_users`. If this list of users
+-- is needed syncronously, this function will need to be refactored.
+-- The list of taggable users should contain:
+--   - The PR author
+--   - The authors of all the existing comments
+--   - The contributors of the repo
+local function async_fetch_taggable_users(bufnr, repo)
+  local users = api.nvim_buf_get_var(bufnr, "taggable_users") or {}
+  local comments_metadata = api.nvim_buf_get_var(bufnr, "comments")
+  for _, c in pairs(comments_metadata) do
+    table.insert(users, c.author)
+  end
+  api.nvim_buf_set_var(bufnr, "taggable_users", users)
+  gh.run(
+    {
+      args = {"api", format("repos/%s/contributors", repo)},
+      cb = function(response)
+        local resp = json.parse(response)
+        for _, contributor in ipairs(resp) do
+          table.insert(users, contributor.login)
+        end
+        api.nvim_buf_set_var(bufnr, "taggable_users", users)
+      end
+    }
+  )
+end
+
+-- This function fetches the issues in the repo so they can be used for
+-- completion.
+local function async_fetch_issues(bufnr, repo)
+  gh.run(
+    {
+      args = {"api", format(format("repos/%s/issues", repo))},
+      cb = function(response)
+        local resp = json.parse(response)
+        api.nvim_buf_set_var(bufnr, "issues", resp)
```

Comment:
I'm putting the whole issue response in here which we definitely don't need. Would you like me to build smaller tables, similar to what happens with the `comments_metadata`? We probably actually only need title and number.

Reply:
I went ahead and just made that change since there was so much that wasn't needed :shrug: